ddspp.h

```diff
@@ -1005,32 +1005,38 @@ namespace ddspp
 
 		if (desc.type == Texture3D)
 		{
-			for (unsigned int m = 0; m < mip; ++m)
+			for (unsigned int m = 0; m <= mip; ++m)
 			{
-				unsigned long long mipSize = mip0Size >> 2 * m;
-				offset += mipSize * desc.numMips;
+				unsigned int mipWidth = (desc.width >> m) > 1 ? (desc.width >> m) : 1;
+				unsigned int mipHeight = (desc.height >> m) > 1 ? (desc.height >> m) : 1;
+				unsigned int mipBlocksWide = (mipWidth + desc.blockWidth - 1) / desc.blockWidth;
+				unsigned int mipBlocksTall = (mipHeight + desc.blockHeight - 1) / desc.blockHeight;
+				unsigned long long mipSize = mipBlocksWide * mipBlocksTall * desc.bitsPerPixelOrBlock;
+				offset += mipSize * ((m == mip) ? slice : desc.numMips);
```

Comment:
I guess this was to avoid the boilerplate below, I generally don't like these but I guess it's ok